### ISSUE TITLE
[FEA] Refine training benchmark model configuration.

### ISSUE THREAD

**Issue by JacoCheung** [NVIDIA/recsys-examples#121]
Currently, our benchmark configuration can not reflect real scenarios well. 
1. The batchsize or sequence length is not typical
2. Input data is uniform, we need powerlaw.
3. The embedding dim is equal to hidden dim 
4. The attention hyper-params is not well set
5. The model weight is not large enough. (because of 3. and 4.)
7. We have only single node test. 
8. We do not benchmark the TP.
Therefore we need to improve our benchmark suites. 
-----
By submitting this issue, you agree to follow our [code of conduct](https://docs.rapids.ai/resources/conduct/) and our [contributing guidelines](https://github.com/jarmak-nv/rapids-repo-template/blob/main/CONTRIBUTING.md).

**Comment by JacoCheung** [comment]
Hi @shijieliu . I would like issue #117  to address hstu layer perf only. We can move to this issue regarding model perf benchmark.


> usually should be 1k - 8k

I think batchsize might be coupled with sequence length. If we use larger hidden_size and batchsize X seqlen. It may lead to OOM easily. In that sense, I think we need to add large scaling (like 16x8 GPUs) benchmark (or with real datasets). 

>  I think we can use larger hidden_size 
You mean we can use more neads or larger head_dim?

Does this benchmark reflect embedding as well? (i.e. use small dense but intensive dyanmic embedding).

**Comment by JacoCheung** [comment]
Adding a MTMS (multi-thread,multi-stream) pipeline. [branch](https://github.com/NVIDIA/recsys-examples/compare/main...JacoCheung:recsys-examples:junzhang/multi-thread-multi-stream-pipeline)  

Which also supports exposed time measurement.

**Comment by shijieliu** [comment]
close this issue and will track optimization on caching, prefetching and pipeline in new issue @JacoCheung
